integration/src/test/java/com/aws/greengrass/mqttbroker/MQTTServiceTest.java

```diff
@@ -73,11 +85,24 @@ boolean isListeningOnPort(int port) {
     }
 
     @Test
-    void GIVEN_defaultConfig_WHEN_startComponent_THEN_brokerStartsOnPort8883() throws InterruptedException {
+    void GIVEN_defaultConfig_WHEN_startComponent_THEN_brokerStartsOnPort8883()
+        throws InterruptedException, ServiceLoadException, CsrProcessingException, KeyStoreException {
         startNucleusWithConfig("config.yaml");
+        ArgumentCaptor<Consumer<X509Certificate>> captor = ArgumentCaptor.forClass(Consumer.class);
```

Comment:
Separate test?

Reply:
I like to reuse tests when it fits in reasonably enough, as with this case. It saves us on the Nucleus startup time since we can accomplish more during a single test while not reducing the debuggability.

Reply:
I updated the test name at least to reflect that we're testing more.